mapit/models.py

```diff
@@ -47,21 +48,26 @@ def new(self):
         return latest[0]
 
     def query_args(self, request, format):
-        try:
-            generation = int(request.GET.get('generation', 0))
-        except ValueError:
-            raise ViewException(format, _('Bad generation specified'), 400)
-        if not generation:
-            generation = self.current().id
-
-        try:
-            min_generation = int(request.GET.get('min_generation', 0))
-        except ValueError:
-            raise ViewException(format, _('Bad min_generation specified'), 400)
-        if not min_generation:
-            min_generation = generation
+        args = {}
+        for q in ('generation', 'min_generation', 'max_generation'):
+            try:
+                args[q] = int(request.GET.get(q, 0))
+            except ValueError:
+                raise ViewException(format, _('Bad generation specified'), 400)
```

Comment:
This error would be more helpful if it mentioned which param had the bad value. Maybe something like:
```suggestion
                raise ViewException(format, _('Bad value specified for %s parameter') % q, 400)
```